### DISCUSSION TITLE
How do I set audio stream's frame_size?

### DISCUSSION THREAD

**Discussion by makerjackie** [PyAV-Org/PyAV#852] [1. Help]
## Overview

I want to get AudioFrame with duration 20ms (160 samples / sample rate 8000 ) from a wavfile.

but with code
```
container = av.open('8k.wav')
for frame in container.decode(audio=0):
    print(frame.samples)
```
I get `2048` which is 0.256 ms

## Research
* I found a `stream_option` in [av.open docs](https://pyav.org/docs/develop/api/_globals.html) but have no idea how to use it
* I found [av.audio.codeccontext.AudioCodecContext.frame_size](https://pyav.org/docs/develop/api/audio.html?highlight=audioplane#av.audio.codeccontext.AudioCodecContext.frame_size), but I don't know how to change it too (I get `0` when I print it)

I have done the following:

- [x] Checked the [PyAV documentation](https://pyav.org/docs)
- [x] Searched on [Google](https://www.google.com/search?q=pyav+how+do+I+foo)
- [x] Searched on [Stack Overflow](https://stackoverflow.com/search?q=pyav)
- [x] Looked through [old GitHub issues](https://github.com/PyAV-Org/PyAV/issues?&q=is%3Aissue)
- [x] Asked on [PyAV Gitter](https://gitter.im/PyAV-Org)
- [x] ... and waited 72 hours for a response.

**Comment by RenaKunisaki** [comment]
I'm also seeing `frame_size` set to zero. Bug?

**Reply by pengzhendong** [reply]
Not bug. It means variable frame size.

https://github.com/FFmpeg/FFmpeg/blob/30996b71950317feae506a8abf87af286f3cedd3/libavcodec/avcodec.h#L1045-L1051

```
Each submitted frame except the last must contain exactly frame_size samples per channel.
May be 0 when the codec has AV_CODEC_CAP_VARIABLE_FRAME_SIZE set, then the frame size is not restricted.
```
